### PR TITLE
Add DurableQueue subscribe option for RabbitMQ broker

### PR DESCRIPTION
**Pull request by Taik**

### CODE REVIEW

File: broker/rabbitmq/connection.go [hunk 1]
```diff
@@ -136,13 +136,19 @@ func (r *rabbitMQConn) tryToConnect(secure bool, config *tls.Config) error {
 	return nil
 }
 
-func (r *rabbitMQConn) Consume(queue, key string, autoAck bool) (*rabbitMQChannel, <-chan amqp.Delivery, error) {
+func (r *rabbitMQConn) Consume(queue, key string, autoAck bool,
+	declareDurableQueue bool) (*rabbitMQChannel, <-chan amqp.Delivery, error) {
```
Comment (asim) [inline broker/rabbitmq/connection.go]: Please change this to `Consume(queue, key string, autoAck, durableQueue bool)` on one line

Comment (Taik) [inline broker/rabbitmq/connection.go]: Done!

File: broker/rabbitmq/connection.go [hunk 2]
```diff
@@ -136,13 +136,18 @@ func (r *rabbitMQConn) tryToConnect(secure bool, config *tls.Config) error {
 	return nil
 }
 
-func (r *rabbitMQConn) Consume(queue, key string, autoAck bool) (*rabbitMQChannel, <-chan amqp.Delivery, error) {
+func (r *rabbitMQConn) Consume(queue, key string, autoAck, declareDurableQueue bool) (*rabbitMQChannel, <-chan amqp.Delivery, error) {
```
Comment (asim) [inline broker/rabbitmq/connection.go]: Please change it to "durableQueue" as mentioned previously.

File: broker/rabbitmq/rabbitmq.go [hunk 3]
```diff
@@ -75,7 +75,12 @@ func (r *rbroker) Subscribe(topic string, handler broker.Handler, opts ...broker
 		o(&opt)
 	}
 
-	ch, sub, err := r.conn.Consume(opt.Queue, topic, opt.AutoAck)
+	ch, sub, err := r.conn.Consume(
+		opt.Queue,
+		topic,
+		opt.AutoAck,
+		opt.Context.Value(durableQueueKey{}).(bool),
```
Comment (asim) [inline broker/rabbitmq/rabbitmq.go]: Will this not panic if there's a nil value?

File: broker/rabbitmq/connection.go (hunk 2 above)
Comment (Taik) [inline broker/rabbitmq/connection.go]: Apologies, I didn't see it changed--fixed.

File: broker/rabbitmq/rabbitmq.go (hunk 3 above)
Comment (Taik) [inline broker/rabbitmq/rabbitmq.go]: Thanks for the catch--please see the latest commit.

**Pull request merged**